### PR TITLE
Make Sky initialize shaders only once

### PR DESCRIPTION
**Pull request by reduz**
* Create the shader on demand.
* Makes sure compilation happens only once, then shader is cached.
* Speeds up scene loading times.
* Speeds up editor initialization.

### CODE REVIEW

File: scene/register_scene_types.cpp [hunk 1]
```diff
@@ -1085,6 +1085,10 @@ void unregister_scene_types() {
 	BaseMaterial3D::finish_shaders();
 #endif // _3D_DISABLED
 
+	PhysicalSkyMaterial::cleanup_shader();
+	PanoramaSkyMaterial::cleanup_shader();
+	ProceduralSkyMaterial::cleanup_shader();
```
Comment (akien-mga) [inline scene/register_scene_types.cpp:1090]: Should they be named `finish_shaders` to what other classes do?

**Comment by akien-mga** [comment]
There's a crash on CI with the sanitizers build and test project:
```
servers/rendering/renderer_scene_cull.cpp:3841:75: runtime error: member access within null pointer of type 'struct RendererSceneOcclusionCull'
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on godotengine/godot/issues
[1] bin/godot.linuxbsd.tools.64s() [0x2100956] (/home/runner/work/godot/godot/platform/linuxbsd/crash_handler_linuxbsd.cpp:54)
[2] /lib/x86_64-linux-gnu/libc.so.6(+0x46210) [0x7fbcdad56210] (??:0)
[3] RendererSceneCull::free(RID) (/home/runner/work/godot/godot/servers/rendering/renderer_scene_cull.cpp:3841)
[4] RenderingServerDefault::_free(RID) (/home/runner/work/godot/godot/servers/rendering/rendering_server_default.cpp:57)
[5] RenderingServerDefault::free(RID) (/home/runner/work/godot/godot/servers/rendering/rendering_server_default.h:867)
[6] PhysicalSkyMaterial::cleanup_shader() (/home/runner/work/godot/godot/scene/resources/sky_material.cpp:527)
[7] unregister_scene_types() (/home/runner/work/godot/godot/scene/register_scene_types.cpp:1089 (discriminator 2))
[8] Main::cleanup(bool) (/home/runner/work/godot/godot/main/main.cpp:2660)
[9] bin/godot.linuxbsd.tools.64s(main+0x418) [0x20fe28e] (/home/runner/work/godot/godot/platform/linuxbsd/godot_linuxbsd.cpp:62)
[10] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7fbcdad370b3] (??:0)
[11] bin/godot.linuxbsd.tools.64s(_start+0x2e) [0x20fddbe] (??:?)
-- END OF BACKTRACE --
```

File: scene/register_scene_types.cpp (hunk 1 above)
Comment (reduz) [reply scene/register_scene_types.cpp:1090]: They dont do exacly the same I think

**Review by clayjohn — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**